common4j/src/main/com/microsoft/identity/common/java/authorities/Authority.java

```diff
@@ -134,7 +135,12 @@ public static Authority getAuthorityFromAuthorityUrl(String authorityUrl) {
 
         final List<String> pathSegments = authorityCommonUriBuilder.getPathSegments();
 
-        if (pathSegments.size() == 0) {
+        // Adding check in case we have a trailing "/" at the end of the authority
+        if (pathSegments.size() == 0 || (pathSegments.size() == 1 && pathSegments.get(0).equals(""))) {
+            if (authorityUrl.contains("ciamlogin.com")){
```

Comment:
Why do we need to do this check? Shouldn't CIAM authorities by always known from configuration?

Reply:
We would have to move the path segment check to a point after the `authorityIsKnownFromConfiguration` call. I think this is fine, but didn't want to mess with the logic of this method too much as it's used in pretty much every MSAL flow. I think it's safer to keep it as is for now, and once we've validated the current changes, we can go back and re-order this method. Also I'd probably still keep this second condition incase the shortened CIAM Authority is not known to MSAL for some reason.

Reply:
I think this is a question about how is the feature supposed to work? Should developers be allowed to get a token from a CIAM Authority even if the authority is not declared in msal json configuration?